src/expr/macro/src/gen.rs

```diff
@@ -115,12 +115,6 @@ impl FunctionAttr {
                 // infer as the type of "anymap" argument
                 return Ok(quote! { |args| Ok(args[#i].clone()) });
             }
-        } else if self.ret == "vector" {
-            if let Some(i) = self.args.iter().position(|t| t == "vector") {
-                // infer as the type of "vector" argument
-                // Example usage: `last_value(*) -> auto`
-                return Ok(quote! { |args| Ok(args[#i].clone()) });
-            }
```

Comment:
The hidden special rules for array/structmap are better removed and made explicit as well.